quarkus-integration/quarkus/devui-integration-test/src/test/java/ai/timefold/solver/quarkus/it/devui/TimefoldDevUIMultipleSolversTest.java

```diff
@@ -0,0 +1,111 @@
+
+import io.quarkus.devui.tests.DevUIJsonRPCTest;
+import io.quarkus.test.QuarkusDevModeTest;
+
+public class TimefoldDevUIMultipleSolversTest extends DevUIJsonRPCTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .setBuildSystemProperty("quarkus.timefold.solver.\"solver1\".environment-mode", "FULL_ASSERT")
+            .setBuildSystemProperty("quarkus.timefold.solver.\"solver2\".environment-mode", "REPRODUCIBLE")
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(StringLengthVariableListener.class,
+                            TestdataStringLengthShadowEntity.class, TestdataStringLengthShadowSolution.class,
+                            TestdataStringLengthConstraintProvider.class, TimefoldTestMultipleResource.class));
+
+    @Path("/timefold/test")
+    public static class TimefoldTestMultipleResource {
+
+        @Inject
+        @Named("solver1")
+        SolverManager<TestdataStringLengthShadowSolution, Long> solverManager;
+
+        @Inject
+        @Named("solver2")
+        SolverManager<TestdataStringLengthShadowSolution, Long> solverManager2;
+    }
+
+    public TimefoldDevUIMultipleSolversTest() {
+        super("Timefold Solver");
+    }
+
+    @Test
+    void testSolverConfigPage() throws Exception {
+        JsonNode configResponse = super.executeJsonRPCMethod("getConfig");
+        List.of(Pair.of("solver1", "FULL_ASSERT"), Pair.of("solver2", "REPRODUCIBLE")).forEach(key -> {
+            String solverConfig = configResponse.get("config").get(key.getLeft()).asText();
+            assertThat(solverConfig).isEqualToIgnoringWhitespace(
+                    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                            + "<!--Properties that can be set at runtime are not included-->\n"
+                            + "<solver>\n"
+                            + "  <environmentMode>" + key.getRight() + "</environmentMode>\n"
+                            + "  <solutionClass>" + TestdataStringLengthShadowSolution.class.getCanonicalName()
+                            + "</solutionClass>\n"
+                            + "  <entityClass>" + TestdataStringLengthShadowEntity.class.getCanonicalName() + "</entityClass>\n"
+                            + "  <domainAccessType>GIZMO</domainAccessType>\n"
+                            + "  <scoreDirectorFactory>\n"
+                            + "    <constraintProviderClass>" + TestdataStringLengthConstraintProvider.class.getCanonicalName()
+                            + "</constraintProviderClass>\n"
+                            + "  </scoreDirectorFactory>\n"
+                            + "  <termination/>\n"
+                            + "</solver>");
+        });
+    }
+
+    @Test
+    void testModelPage() throws Exception {
+        JsonNode modelResponse = super.executeJsonRPCMethod("getModelInfo");
+        List.of("solver1", "solver2").forEach(key -> {
```

Comment:
For sake of simplicity, I'd refactor this to avoid the lambda.
Make the assertions be inside a method, have the string as an argument to the method, call the method twice.

The stack trace in case of a failure will be nicer, and more self-explanatory.